### PR TITLE
Update gemspec dependencies for ruby 3.3

### PR DESCRIPTION
**Pull request by kbrock**
ruby 3.3 deprecation warnings

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,14 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased] - undecided
 
+## [1.8.1] - 2024-03-01
+
+### Added
+
+- Explicitly added ostruct dependency since it is leaving the standard library.
+- Explicitly added csv dependency. It is required by prawn-table.
+- Explicitly added matrix. It is required by prawn. (We're not ready to upgrade to 2.5 yet)
```
Comment (jrafanie) [inline CHANGELOG.md:15]: @kbrock for csv and matrix, why do we need to add indirect dependencies as direct dependencies?  Is there a deprecation warning this addresses?

I would have thought prawn-table and prawn should state their dependencies.  What am I missing?

Comment (kbrock) [reply CHANGELOG.md:15]: @jrafanie prawn-table hasn't been updated in years, and I am pessimistic that we'll get this change in there.

Prawn has been updated. The best approach is to upgrade the dependency but it was failing the build so I took the easy way out.

https://github.com/prawnpdf/prawn/commit/3658d5125c3b20eb11484c3b039ca6b89dc7d1b7

I created a PR for prawn-table

 https://github.com/prawnpdf/prawn-table/pull/161

Comment (jrafanie) [reply CHANGELOG.md:15]: Can you fixup the commit message with that info?  Note the typo too:
`Ostrinct dependency now explicit`.  I'm not sure what `declaring` means for the other commit.

Comment (jrafanie) [reply CHANGELOG.md:15]: I'm ok with adding indirect dependencies to correct a problem in a direct dependency but we should make this very obvious so we can undo this when/if this is ever fixed.  Please comment on the PRs so we can follow up sometime later to see if they have fixed and released it.

**Comment by kbrock** [comment]
@jrafanie thanks - future me thanks you a lot.

update:
- updated commit message to be much more clear
- updated gemspec to properly state dependencies and when we can remove this indirect dependency.

**Comment by kbrock** [comment]
update:
- fixed the ostruct typo

File: CHANGELOG.md (hunk 1 above)
Comment (jrafanie) [reply CHANGELOG.md:15]: Should we bump the VERSION file to match this entry?  Does it make sense to bump minor version due to the dependency changes? It always feels odd when dependencies change in a patch release.

Comment (jrafanie) [reply CHANGELOG.md:15]: @kbrock Can you bump the VERSION file to 1.8.1?

**Comment by kbrock** [comment]
update:
- update version.rb
- changed date

**Review by jrafanie — APPROVED** [review]
LGTM

**Pull request merged**